pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -216,6 +217,7 @@ public boolean isMutableSegment() {
     _partitionFunction = config.getPartitionFunction();
     _mainPartitionId = config.getPartitionId();
     _nullHandlingEnabled = config.isNullHandlingEnabled();
+    _consumerDir = config.getConsumerDir() != null ? new File(config.getConsumerDir()) : null;
```

Comment:
Can this ever be `null`? Currently the abstraction is not very clear, and I don't really see proper null handling. Can we ensure it is always set?

Reply:
That's a good point, it can be null in the `MemoryEstimator` path. I think it's probably clearer to set a dummy dir there, and remove the `@Nullable` annotation for consumerDir

For the realtime flow `RealtimeTableDataManager.getConsumerDir()` ensures it will not be null.